### DISCUSSION TITLE
Understanding Node.js dependency and collaboration on GitHub

### DISCUSSION THREAD

**Discussion by youngdjn** [thuliteio/doks#744] [Q&A]
Hi, I'm hoping to understand how the Node.js dependency affects collaboration with multiple website authors on GitHub. It seems the files installed by Node/npm remain local (they appear to be in the `.gitignore`). So when my collaborators clone the website repo, do they need to install Node/npm as well? Is that all that is necessary for them to begin collaborating?

Relatedly, I don't understand why the Node/npm files are even necessary given they remain local, and Netlify is able to build the site without them.

Thanks for helping a web design newbie who's clueless Node and npm! And thanks for the great theme!

**Comment by h-enk** [comment]
> Thanks for helping a web design newbie who's clueless Node and npm! And thanks for the great theme!

That's totally fine + great to hear. I'm in the process of rewriting a three piece I wrote on npm: "Three reasons why you should use npm with Hugo", "How to manage dependencies with npm — for Hugo" , and "How to customize scripts with npm — for Hugo".  That should clarify a lot more, I will let you know when it's available.

> It seems the files installed by Node/npm remain local (they appear to be in the .gitignore).

Correct.

> So when my collaborators clone the website repo, do they need to install Node/npm as well?

Yes, they do  — to be able to do a build. Note that, content is not fully separated from a build process — e.g. CSS gets purged based on content a/o.

> Is that all that is necessary for them to begin collaborating?

Your collaborators would also need [git](https://git-scm.com/) and access to the repository where the code lives. Git (in combination w/ GitHub) is how you'd collaborate basically — see also GitHub's [Git Guide](https://github.com/git-guides). Note that if your collaborators would only need to be able to update content, a Content Management System would come in handy (and access to it is then only needed).

> Relatedly, I don't understand why the Node/npm files are even necessary given they remain local, and Netlify is able to build the site without them.

The dependencies are needed (and Netlify does install + uses them) to generate a static build.
